### PR TITLE
Fetch data from the spacexdata API

### PR DESCRIPTION
**Pull request by Terbeche**
## Hi @RitobrotoMukherjee  I've fetched the rockets data from the spacexdata API 
I fetched data from the Rockets [endpoint](https://api.spacexdata.com/v3/rockets) when the application starts

Once the data are fetched, I dispatched an action to store the selected data in Redux store:
- id
- name
- description
- flickr_images

Use useSelector() Redux Hook to select the state slices and render lists of rockets in corresponding routes.

### CODE REVIEW

File: src/components/Rockets.js [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+import { fetchRockets } from '../redux/rockets/rockets';
 
-const Rockets = () => (
-  <></>
-);
+const Rockets = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchRockets());
```
Comment (RitobrotoMukherjee) [inline src/components/Rockets.js:10]: @Terbeche in the `useEffect` this `dispatch` is running every-time the component **mounts** or **updates**. Thus `store` is also getting updated every-time. The requirement is to load the **rocket data** once and store it in the `Redux Store`.
To achieve this you can check that, if **rockets** data is there in the store then don't `dispatch` the `thunk` function.

File: src/redux/rockets/rockets.js [hunk 2]
```diff
@@ -1,4 +1,45 @@
-const rockets = () => (
-  <></>
-);
-export default rockets;
+/* eslint-disable */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { toHaveDescription } from '@testing-library/jest-dom/dist/matchers';
```
Comment (RitobrotoMukherjee) [inline src/redux/rockets/rockets.js:3]: This line is not used anywhere, please remove it to follow the `YAGNI` principle in your code.

**Review by RitobrotoMukherjee — CHANGES_REQUESTED** [review]
## Hi @Terbeche, 
Great job implementing the requirements.

**Key highlights:**
- No linters errors. 👍 
- Descriptive pull request. 💯 
- Good code structures. 🥇

But there are still few changes needed to finish this project. Please consider the comments below. 🔢 
**Happy Coding 💻**

File: src/components/Rockets.js (hunk 1 above)
Comment (Terbeche) [reply src/components/Rockets.js]: Thank you @RitobrotoMukherjee for the feedback, it's done now.

File: src/redux/rockets/rockets.js (hunk 2 above)
Comment (Terbeche) [reply src/redux/rockets/rockets.js]: @RitobrotoMukherjee Thanks for the remarks, I fixed it.

**Review by RitobrotoMukherjee — APPROVED** [review]
### Hi @Terbeche ,
You have done an awesome job. **There is nothing else to say other than, it's time to merge.**

### Key Highlights:
- No linters errors.
- Resolved all previously mentioned requirements.
- Good design.

giphy

**Happy Coding**

**Pull request merged**